ast-nodes/if-statement.dart

```diff
@@ -10,30 +10,56 @@ class IfStatement implements Statement {
 
   IfStatement(this.condition, this.blockTrue, this.blockFalse);
 
-  // TODO: implement .parse()
-
   factory IfStatement.parse(Iterable<Token> tokens) {
     var iterator = tokens.iterator;
     checkNext(iterator, RegExp('if\$'), "Expected 'if'");
     var expressionBody = consumeUntil(iterator, RegExp("^then\$"));
     checkThis(iterator, RegExp('then\$'), "Expected 'then'");
-    var body = consumeUntil(iterator, RegExp("^else\$"));
-    checkThis(iterator, RegExp('else\$'), "Expected 'else'");
-    var elseBody = consumeUntil(iterator, RegExp("^end\$"));
-    checkThis(iterator, RegExp('end\$'), "Expected 'end'");
-    checkNoMore(iterator);
+    var trueBody = consumeUntil(iterator, RegExp("^(else|end)\$"));
+    List<Token> falseBody = null;
+
+    if (iterator.current?.value == "else") {
+
+      falseBody = consumeUntil(iterator, RegExp("^end\$"));
+      checkThis(iterator, RegExp('end\$'), "Expected 'end'");
+      checkNoMore(iterator);
```

Comment:
We can have another if else end statement inside, so using consumeUntil may wrongly parse it. Try to use consumeStackUntil(iterator, for | if, end), or write your own cycle for checking.

If smth then
else
  if smth then
  then
  end
end

I mean this structure